### PR TITLE
Edits for clarifying process, visual presentation, and removal of ina…

### PR DESCRIPTION
**Pull request by Adorism**
…ccuracies as well as the removal of sections that will be inserted elsewhere in documentation

## What
These edits improve the introduction to the overview of building new connectors. 
I've removed material that is no longer accurate. 
I've removed instructions specifically addressing the process for contributions, instead directing readers to the documents on contributing newly built connectors. 
I've reorganized some of the content into a table to enhance clarity. 

## How
Docs Update

## Review guide
Please keep in mind that I'll be finding a more user-friendly location in the docs for deleted information on "Updating existing connectors" and "Using credentials in UI"

## User Impact
Removes ambiguity of text relating to user intent. Previously, this doc was addressing two different audiences with different needs.
1) Those who want to contribute a connector to the catalog. 
2) Those who want to build a connector specifically for use in their own workspace, with no need to read instructions that only apply to contributors.

This doc is now more clearly applicable to anyone looking to build a connector and guides those with more specific needs to the next applicable piece of documentation. 

## Can this PR be safely reverted and rolled back?

- [X] YES 💚
- [ ] NO ❌

### CODE REVIEW

File: docs/connector-development/README.md [hunk 1]
```diff
@@ -1,158 +1,33 @@
...
-access to it.
+If you need support along the way, visit the [Slack channel](https://airbytehq.slack.com/archives/C027KKE4BCZ) we have dedicated to helping users with connector development where you can search previous discussions or ask a question of your own. 
 
-### Connector Builder
+### Process overview
 
-The [connector builder UI](connector-builder-ui/overview.md) is based on the low-code development
-framework below and allows to develop and use connectors without leaving the Airbyte web UI. No
-local developer environment required.
+The first step in creating a new connector is to choose the tools you’ll use to build it. There are three basic approaches Airbyte provides to start developing a connector. To understand which approach you should take, review the [compatibility guide](./connector-builder-ui/connector-builder-compatibility.md).  [The Airbyte community also maintains some CDKs](#community-maintained-cdks) suitable for certain use cases.
```
Comment (natikgadzhi) [inline docs/connector-development/README.md:11]:
```suggestion
The first step in creating a new connector is to choose the tools you’ll use to build it. There are three basic approaches Airbyte provides to start developing a connector. To understand which approach you should take, review the [compatibility guide](./connector-builder-ui/connector-builder-compatibility.md).
```

File: docs/connector-development/README.md [hunk 2]
```diff
@@ -1,158 +1,33 @@
...
 
-### Low-code Connector-Development Framework
+After building and testing your connector, you’ll need to publish it. This makes it available in your workspace. At that point, you can use the connector you’ve built to move some data! 
 
-You can use the [low-code framework](config-based/low-code-cdk-overview.md) to build source
-connectors for HTTP API sources. Low-code CDK is a declarative framework that provides a YAML schema
-to describe your connector without writing any Python code, but allowing you to use custom Python
-components if required.
+If you want to contribute what you’ve built to the catalog of publicly available connectors, follow the steps provided in the [contribution guide for submitting new connectors](../contributing-to-airbyte/submit-new-connector.md).
```
Comment (natikgadzhi) [inline docs/connector-development/README.md:15]:
```suggestion
If you want to contribute what you’ve built to the Airbyte Cloud and OSS connector catalog, follow the steps provided in the [contribution guide for submitting new connectors](../contributing-to-airbyte/submit-new-connector.md). 
```

**Review by natikgadzhi — APPROVED** [review]
@Adorism thank you! <3 Made a loom reviewing this: https://www.loom.com/share/1be3582034b74ac7a5a2f65e46e20c74

Already great, suggested a couple of changes.

**Review by marcosmarxm — APPROVED** [review]
Clean and direct to the point, awesome!

**Pull request merged**